lib/map.h

```diff
@@ -17,42 +17,66 @@ limitations under the License.
 #ifndef LIB_MAP_H_
 #define LIB_MAP_H_
 
-#include <map>
+#include <optional>
 
-template <class K, class T, class V, class Comp, class Alloc>
-inline V get(const std::map<K, V, Comp, Alloc> &m, T key, V def = V()) {
+/// Given a map and a key, return the value corresponding to the key in the map,
+/// or a given default value if the key doesn't exist in the map.
+template <typename Map, typename Key>
+typename Map::mapped_type get(const Map &m, const Key &key) {
     auto it = m.find(key);
-    if (it != m.end()) return it->second;
-    return def;
+    return it != m.end() ? it->second : typename Map::mapped_type{};
+}
+template <class Map, typename Key = typename Map::key_type,
```

Comment:
Why has the Key default type here, but not in the previous overload?

Reply:
Actually an oversight. I was playing with heterogenous lookup and this is just a leftover. Can be removed, we have `mapped_type` to distinguish map types from everything else.